src/query/arithmetic.c

```diff
@@ -5241,6 +5241,50 @@ db_json_pretty_dbval (DB_VALUE * json, DB_VALUE * res)
     }
 }
 
+int
+db_json_arrayagg_dbval (DB_VALUE * json, DB_VALUE * json_res)
+{
+  JSON_DOC *this_doc;
+  const char *raw_path;
+  JSON_DOC *result_doc = NULL;
+  int error_code = NO_ERROR;
+
+  if (DB_IS_NULL (json))
```

Comment:
I don't think result should be made NULL when an aggregated value is NULL.

Reply:
This should not happen in the first place. If we encounter a NULL value before calling the function, we will make the corresponding db_value a JSON with the type DB_JSON_NULL. Therefore this condition should not be evaluated to true. I think here we can put an `assert (false)`, call `db_make_null (json_res)` and `return ER_FAILED`